httpserver/service_node.cpp

```diff
@@ -401,9 +398,9 @@ void ServiceNode::bootstrap_data() {
     }
 }
 
-bool ServiceNode::snode_ready(boost::optional<std::string&> reason) {
+bool ServiceNode::snode_ready(std::string* reason) {
```

Comment:
Anything wrong about using an optional for a reference here?

Reply:
Yes, the C++ standard committee rejected std::optional support for reference types (at least for C++17) because it had a lot of controversy on what it should do (rebind or fall-through).  The C++17 alternatives are either a pointer (like this), or a std::reference_wrapper.  In most cases (like here) a nullable pointer does the job just fine.